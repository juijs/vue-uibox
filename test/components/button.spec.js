import { mount } from 'vue-test-utils'
import UixButton from '@/components/button'

describe('button.js', () => {
    const wrapper = mount(UixButton, {
        propsData: {
            items: [
                { icon: "", text: "A", value: "1", style: { width: "70px" } },
                { icon: "", text: "B", value: "2", style: { width: "70px" } },
                { icon: "", text: "C", value: "3", style: { width: "70px" } },
                { icon: "home", text: "Home", value: "home", focus: true },
                { icon: "gear", text: "Gear", value: "gear" },
                { icon: "help", text: "", value: "help" }
            ]
        }
    });

    it('renders li for each item in props.items', () => {
        expect(wrapper.vm.items.length).toBe(6);
    })

    it('server render', () => {
        wrapper.vm.$nextTick(() => {
            expect(wrapper.vm.$el).toMatchSnapshot();
        });
    });
})