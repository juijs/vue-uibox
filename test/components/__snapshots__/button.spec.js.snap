// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button.js server render 1`] = `
<div
  class="group"
>
  <a
    class="btn small active"
    style="width: 70px;"
    value="1"
  >
    <!---->
    
        A
    
  </a>
  <a
    class="btn small"
    style="width: 70px;"
    value="2"
  >
    <!---->
    
        B
    
  </a>
  <a
    class="btn small"
    style="width: 70px;"
    value="3"
  >
    <!---->
    
        C
    
  </a>
  <a
    class="btn small focus"
    value="home"
  >
    <i
      class="icon-home"
    />
    
        Home
    
  </a>
  <a
    class="btn small"
    value="gear"
  >
    <i
      class="icon-gear"
    />
    
        Gear
    
  </a>
  <a
    class="btn small"
    value="help"
  >
    <i
      class="icon-help"
    />
    
        
    
  </a>
</div>
`;
